src/main/database.cpp

```diff
@@ -68,6 +84,21 @@ void Database::setLoggingLevel(std::string loggingLevel) {
     spdlog::set_level(LoggingLevelUtils::convertStrToLevelEnum(std::move(loggingLevel)));
 }
 
+void Database::checkAccessMode() {
+    if (!FileUtils::fileOrPathExists(databasePath)) {
+        FileUtils::createDir(databasePath);
+    }
```

Comment:
Remove L88 to L90 after you move this function into `initDBDirAndCoreFilesIfNecessary`.